src/herder/HerderImpl.cpp

```diff
@@ -977,7 +984,16 @@ HerderImpl::restoreUpgrades()
     {
         Upgrades::UpgradeParameters p;
         p.fromJson(s);
-        mUpgrades.setParameters(p);
+        try
+        {
+            // use common code to set status
```

Comment:
I think this comment is misleading. What status?

Reply:
it refers to `statusManager`, I had to put this comment to explain why it makes sense to do read followed by write. I thought that splitting status out of `setUpgrades` was not worth it as it would create code paths where people forget to update status